File: msg-entity/src/main/java/dev/number6/entity/dagger/ChannelMessagesEntityComprehensionComponent.java
package dev.number6.entity.dagger;

import dev.number6.CommonModule;
import dev.number6.message.ChannelMessagesHandler;
import dagger.Component;

@Component(modules = {
        CommonModule.class,
        ChannelMessagesHandlerModule.class,
        ComprehensionResultsModule.class
})
public interface ChannelMessagesEntityComprehensionComponent {

    ChannelMessagesHandler getChannelMessagesHandler();
}

File: msg-entity/src/main/kotlin/dev/number6/entity/dagger/ChannelMessagesEntityComprehensionComponent.kt
package dev.number6.entity.dagger

import dagger.Component
import dev.number6.ComprehendModule
import dev.number6.message.ChannelMessagesHandler

@Component(modules = [ComprehendModule::class, ChannelMessagesHandlerModule::class, ComprehensionResultsModule::class])
interface ChannelMessagesEntityComprehensionComponent {
    fun getChannelMessagesHandler(): ChannelMessagesHandler
}